### DISCUSSION TITLE
How to change the highlight color for comments.

### DISCUSSION THREAD

**Discussion by harryhanYuhao** [NvChad/NvChad#2722] [Q&A] [answered]
I am using onedark color theme. I think its highlight for the comments are too dark to see.

 I have tried to overwrite the highlight color but have not succeeded. 

Here is what I have tried:

```lua
-- file lua/chadrc.lua
M.ui = {
  changed_themes = {
      onedark = {
         base_16 = {
            base04 = "#777777", -- no effects at all
         },
         base_30 = {
            gray_fg="#777777" -- no effects at all
         },
      },
  },
  hl_override = {
    Comment = {
      bg = "#777777", -- no effects at all
      fg = "#777777" -- no effects at all
    }
  },
  theme = "onedark",
}
```

I have looked up the color defined here https://github.com/NvChad/base46/blob/v2.0/lua/base46/themes/onedark.lua
And I believe the color for the comment are defined by `base04` and `gray_fg`.

May anyone tell me how to change the color properly?

**Comment by siduck** [comment, marked as answer]
add `["@comment"]` too, 99% of time everyone uses treesitter syntax highlighting. So its hlgroups must be used too. Always check the treesitter integration file to know correct names

**Reply by kwdevx** [reply]
using `hl_override` will be a global setting for all themes

what if we want to use the original `polish_hl` property in the v2.5 for particular theme

after update, this property seems not loaded

example theme:
```lua
local M = {}

M.base_30 = {
...
}

M.base_16 = {
...
}

M.polish_hl = {
  ["@variable"] = { fg = M.base_30.lavender },

  ["@keyword"] = { fg = M.base_30.cyan },
  Repeat = { fg = M.base_30.cyan },
  Operator = { fg = M.base_30.cyan },
  ["@operator"] = { fg = M.base_30.cyan },

  ["@property"] = { fg = M.base_16.base04 },
  ["@type.builtin"] = { fg = M.base_30.purple },
  Typedef = { fg = M.base_30.yellow },
  Type = { fg = M.base_30.yellow },
  Structure = { fg = M.base_30.yellow },

  ["@punctuation.bracket"] = { fg = M.base_30.cyan },
  -- ["@parenthesis"] = { fg = M.base_30.dark_purple },

  Comment = {
    italic = true,
  },
}

M.type = "dark"

-- M = require("base46").override_theme(M, "palenight")

return M

```

**Reply by siduck** [reply]
@Haircrab you can modify hl_override to configure it per theme, just need some lua knowledge :sunglasses: 

![image](https://github.com/NvChad/NvChad/assets/59060246/770d7840-900a-408f-bc0a-a7bcdab9ed3b)

![image](https://github.com/NvChad/NvChad/assets/59060246/837f4218-4a03-4d7a-ae79-83ef1cb1da10)

**Comment by nattee** [comment]
I was struggling with this for a while and this is how I did it.

Objective: 
* Make comments in any theme be italic
* For `wombat` theme only, change the foreground color of the delimiter to the orange color of the wombat theme (https://github.com/NvChad/base46/blob/v2.5/lua/base46/themes/wombat.lua), which is defined as `#FFCC66`.

Here is how I make it work.

```
-- file lua/chadrc.lua
local M = {}

M.base46 = {
  theme = "wombat",

  hl_override = {
    Comment = { italic = true },
    ["@comment"] = { italic = true },
  },
  changed_themes = {
    wombat = {
      polish_hl = {
        treesitter = {
          ["@punctuation.delimiter"] = { fg = "#FFCC66"},
        }
      }
    }
  }
}


return M
```

Now, the only thing that still bother me is that I cannot directly use the `base_30.orange` colors from the `wombat.lua` theme file. I have tried `["@punctuation.delimiter"] = { fg = "orange"}` but the color `orange` rendered is NOT the same as `#FFCC66` defined in `wombat.lua`. Using `["@punctuation.delimiter"] = { fg = M.base_30.orange}` also obviously does not work.

Another way is to duplicate the `wombat.lua` into another theme file, says `lua/themes/my-wombat.lua` and add `["@punctuation.delimiter"] = { fg = M.base_30.orange}`. This is ok for me but now I have two wombat themes. I prefer the first option, though.

**Reply by siduck** [reply]
@nattee  changed_themes allows only custom hex colors, not base_30 vars! what are you trying to do

**Reply by nattee** [reply]
@siduck Thanks for the awesome plugins!

My bad, I don't know that `changed_themes` cannot uses base_30 vars.

What I was trying to do is to clearly express my intent in the code. I want the later me (that may read the script years from now) to be able to understand that I want to change the foreground color of `wombat`'s `delimiter` to be `wombat`'s `base_30.orange` and I misunderstood that I can use base_30 vars using some syntax or convention that I don't know. 

Looking back, my intent can be easily conveyed by just commenting it (and I just did, pointing back to this thread).

Thanks for replying.

**Reply by siduck** [reply]
@nattee  use hl_override for it!  

![image](https://github.com/user-attachments/assets/d556e5c9-6153-4fb4-bff5-37bb1a2ec912)

![image](https://github.com/user-attachments/assets/a70c845d-092b-4488-9266-939774f93c56)
